node/pkg/celo/by_transaction.go

```diff
@@ -0,0 +1,98 @@
+package celo
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	celo_common "github.com/celo-org/celo-blockchain/common"
+	"github.com/celo-org/celo-blockchain/ethclient"
+	"github.com/certusone/wormhole/node/pkg/celo/abi"
+	"github.com/certusone/wormhole/node/pkg/common"
+	"github.com/certusone/wormhole/node/pkg/vaa"
+	eth_common "github.com/ethereum/go-ethereum/common"
+)
+
+var (
+	// SECURITY: Hardcoded ABI identifier for the LogMessagePublished topic. When using the watcher, we don't need this
+	// since the node will only hand us pre-filtered events. In this case, we need to manually verify it
+	// since ParseLogMessagePublished will only verify whether it parses.
+	logMessagePublishedTopic = celo_common.HexToHash("0x6eb224fb001ed210e379b335e35efe88672a8ce935d981a6896b27ffdf52a3b2")
+)
+
+// MessageEventsForTransaction returns the lockup events for a given transaction.
+// Returns the block number and a list of MessagePublication events.
+func MessageEventsForTransaction(
+	ctx context.Context,
+	c *ethclient.Client,
+	contract celo_common.Address,
+	chainId vaa.ChainID,
+	tx celo_common.Hash) (uint64, []*common.MessagePublication, error) {
+
+	f, err := abi.NewAbiFilterer(contract, c)
+	if err != nil {
+		return 0, nil, fmt.Errorf("failed to create ABI filterer: %w", err)
+	}
+
+	// Get transactions logs from transaction
+	receipt, err := c.TransactionReceipt(ctx, tx)
+	if err != nil {
+		return 0, nil, fmt.Errorf("failed to get transaction receipt: %w", err)
+	}
+
+	// Bail early when the transaction receipt status is anything other than
+	// 1 (success). In theory, this check isn't strictly necessary - a failed
+	// transaction cannot emit logs and will trigger neither subscription
+	// messages nor have log messages in its receipt.
+	//
+	// However, relying on that invariant is brittle - we connect to a lot of
+	// EVM-compatible chains which might accidentally break this API contract
+	// and return logs for failed transactions. Check explicitly instead.
+	if receipt.Status != 1 {
```

Comment:
Maybe I'm weird, but I'd almost make that 1 a constant like `const statusSuccess = 1` at the top of this file so it is exceedingly obvious, even with the helpful comment. I never like [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming))

Reply:
This is copy pasta from ethereum.  If we want to refactor this, we should probably do it comprehensively.